tests/src/unit-deserialization.cpp

```diff
@@ -166,6 +166,21 @@ struct SaxEventLoggerExitAfterStartArray : public SaxEventLogger
 };
 } // namespace
 
+// Passing a NULL pointer to the input adapter violates its NON_NULL attribute which is detected by UBSAN.
+// To still test whether exceptions are thrown, we need to exclude these tests from UBSAN which can only
+// be done with a function attribute. See
+// https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#disabling-instrumentation-with-attribute-no-sanitize-undefined
+#if defined(__clang__)
+    __attribute__((no_sanitize("undefined")))
+#endif
+void test_file_exception()
+{
+    std::FILE* f = std::fopen("nonexisting_file", "r"); // NOLINT(cppcoreguidelines-owning-memory)
```

Comment:
Can you use `std::FILE* f = nullptr;` (again) now that UBSan is disabled?

Reply:
See discussion below.